### PR TITLE
Allow run `wp-cli` commands in dev mode

### PR DESCRIPTION
**Pull request by fluiddot**
Related to https://github.com/Automattic/studio/pull/257#discussion_r1644117766.

## Proposed Changes

- Include the executable path when running the app using the Electron's shell app.

#### Unpackaged app
- Run the command `npm run start`.
- Click "Open in Terminal".
- In the terminal:
    - Run the command `wp`.
    - Run the command `wp help`.
    - Run the command `wp theme list`
    - Run the command `wp eval 'echo WP_CONTENT_DIR;'`
- Observe that the above commands are executed successfully.

#### Packaged app
- Run the command `npm run package`
- Launch the app by navigating to the `out` directory.
- Click "Open in Terminal".
- In the terminal:
    - Run the command `wp`.
    - Run the command `wp help`.
    - Run the command `wp theme list`
    - Run the command `wp eval 'echo WP_CONTENT_DIR;'`
- Observe that the above commands are executed successfully.

#### Unpackaged app
- Run the command `npm run start`.
- Click "Open in Terminal".
- In the terminal:
    - Run the command `wp`.
    - Run the command `wp help`.
    - Run the command `wp theme list`
    - Run the command `wp eval 'echo WP_CONTENT_DIR;'`
- Observe that the above commands are executed successfully.

#### Packaged app
- Run the command `npm run package`
- Launch the app by navigating to the `out` directory.
- Click "Open in Terminal".
- In the terminal:
    - Run the command `wp`.
    - Run the command `wp help`.
    - Run the command `wp theme list`
    - Run the command `wp eval 'echo WP_CONTENT_DIR;'`
- Observe that the above commands are executed successfully.

#### Unpackaged app
- Run the command `npm run start`.
- Click "Open in Terminal".
- In the terminal:
    - Run the command `wp`.
    - Run the command `wp help`.
    - Run the command `wp theme list`
    - Run the command `wp eval 'echo WP_CONTENT_DIR;'`
- Observe that the above commands are executed successfully.

#### Packaged app
- Run the command `npm run package`
- Launch the app by navigating to the `out` directory.
- Click "Open in Terminal".
- In the terminal:
    - Run the command `wp`.
    - Run the command `wp help`.
    - Run the command `wp theme list`
    - Run the command `wp eval 'echo WP_CONTENT_DIR;'`
- Observe that the above commands are executed successfully.

## Pre-merge Checklist

- [x] Have you checked for TypeScript, React or other console errors?

### CODE REVIEW

File: bin/wp [hunk 1]
```diff
@@ -12,8 +12,4 @@ CLI="wp $COMMAND"
 # Mimic core `wp-cli`'s behavior of using `less` for `help` output.
 PAGER=$(if [ -z "$COMMAND" ] || [[ "$COMMAND" == \'help* ]]; then echo "less -R"; else echo "cat"; fi)
 
-if [ -n "$STUDIO_APP_DIRECTORY" ] && [ -n "$STUDIO_EXE_PATH" ]; then
-  "$STUDIO_EXE_PATH" "$STUDIO_APP_DIRECTORY" --cli="$CLI" | $PAGER
```
Comment (fluiddot) [inline bin/wp:16]: This is no longer needed as the `STUDIO_APP_PATH` will contain the right path:
- Development builds (generated by `npm start`): `<PROJECT_PATH>/node_modules/electron/dist/Electron.exe <PROJECT_PATH>`
- Packaged builds (generated by `npm run package`): `<PROJECT_PATH>/out/**/Studio.exe`

File: bin/wp [hunk 2]
```diff
@@ -12,8 +12,4 @@ CLI="wp $COMMAND"
...
-else
-  "$STUDIO_APP_PATH" --cli="$CLI" | $PAGER
-fi
+$STUDIO_APP_PATH --cli="$CLI" | $PAGER
```
Comment (kozer) [inline bin/wp:15]: TIL. I realized that my solution was a bit hacky, but couldn't make it work! Thanks!

**Review by kozer — APPROVED** [review]
Tested it and works as expected ( Linux ).
Nice work!
image

**Review by dcalhoun — APPROVED** [review]
Test planned succeeded on macOS and Windows for me. 🚀 

I did note an error is logged in the development server when running certain commands. It does not appear to disrupt the UX, but I share it nonetheless. Ideally, we could address the logged error. 

For example, running `wp eval 'echo WP_CONTENT_DIR;'` results in the following development server log: 

```
Error: This does not seem to be a WordPress installation.
The used path is: /var/www/html/
Pass --path=`path/to/wordpress` or run `wp core download`.
```

**Comment by fluiddot** [comment]
> I did note an error is logged in the development server when running certain commands. It does not appear to disrupted the UX, but I share it nonetheless. Ideally, we could address the logged error.
> 
> For example, running `wp eval 'echo WP_CONTENT_DIR;'` results in the following development server log:
> 
> ```
> Error: This does not seem to be a WordPress installation.
> The used path is: /var/www/html/
> Pass --path=`path/to/wordpress` or run `wp core download`.
> ```

Good catch. As you shared, the error doesn't seem to disrupt the command execution. Nonetheless, I'll take a deeper look to better understand why those commands display that warning.

**Pull request merged**